cmake/find/llvm.cmake

```diff
@@ -51,47 +35,88 @@ if (ENABLE_EMBEDDED_COMPILER)
             set (LLVM_FOUND 0)
             set (USE_EMBEDDED_COMPILER 0)
         endif ()
-
-        # TODO: fix llvm 8+ and remove:
-        if (LLVM_FOUND AND LLVM_VERSION_MAJOR GREATER 7)
-            message(WARNING "LLVM 8+ not supported yet, disabling.")
-            set (USE_EMBEDDED_COMPILER 0)
-        endif ()
     else()
-        set (LLVM_FOUND 1)
-        set (USE_EMBEDDED_COMPILER 1)
-        set (LLVM_VERSION "7.0.0bundled")
-        set (LLVM_INCLUDE_DIRS
-            ${ClickHouse_SOURCE_DIR}/contrib/llvm/llvm/include
-            ${ClickHouse_BINARY_DIR}/contrib/llvm/llvm/include
-            ${ClickHouse_SOURCE_DIR}/contrib/llvm/clang/include
-            ${ClickHouse_BINARY_DIR}/contrib/llvm/clang/include
-            ${ClickHouse_BINARY_DIR}/contrib/llvm/llvm/tools/clang/include
-            ${ClickHouse_SOURCE_DIR}/contrib/llvm/lld/include
-            ${ClickHouse_BINARY_DIR}/contrib/llvm/lld/include
-            ${ClickHouse_BINARY_DIR}/contrib/llvm/llvm/tools/lld/include)
-        set (LLVM_LIBRARY_DIRS ${ClickHouse_BINARY_DIR}/contrib/llvm/llvm)
+        if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_CURRENT_BINARY_DIR)
+            message(WARNING "Option ENABLE_EMBEDDED_COMPILER is set but LLVM library cannot build if build directory is the same as source directory.")
+            set (LLVM_FOUND 0)
+            set (USE_EMBEDDED_COMPILER 0)
+        elseif (SPLIT_SHARED_LIBRARIES)
+            # llvm-tablegen cannot find shared libraries that we build. Probably can be easily fixed.
+            message(WARNING "Option ENABLE_EMBEDDED_COMPILER is not compatible with SPLIT_SHARED_LIBRARIES. Build of LLVM will be disabled.")
+            set (LLVM_FOUND 0)
+            set (USE_EMBEDDED_COMPILER 0)
+        elseif (NOT ARCH_AMD64)
+            # It's not supported yet, but you can help.
+            message(WARNING "Option ENABLE_EMBEDDED_COMPILER is only available for x86_64. Build of LLVM will be disabled.")
+            set (LLVM_FOUND 0)
+            set (USE_EMBEDDED_COMPILER 0)
+        else ()
+            set (LLVM_FOUND 1)
+            set (USE_EMBEDDED_COMPILER 1)
+            set (LLVM_VERSION "9.0.0bundled")
```

Comment:
Which version we actually use? Some release or random commit from master?

Reply:
Tag `llvmorg-9.0.0` (release)

Reply:
master is 10.0.0 (unreleased) :)